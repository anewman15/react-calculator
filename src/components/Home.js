/*
 eslint-disable no-unused-vars
*/

import React from 'react';
import Button from 'react-bootstrap/Button';

const Home = () => (
  <div className="container container-fluid">
    <h1 className="text-center">Over the Counter Calculator</h1>
    <p className="display-5 my-4">
      This is one of the primitive calculators. Built with
      React. I'll probably improve it over time. Can be used
      for basic calculations using arithmatic oeprations:
    </p>
    <ul className="list-group w-25 mx-auto">
      <li className="list-group-item">Addition</li>
      <li className="list-group-item">Subtraction</li>
      <li className="list-group-item">Multiplication</li>
      <li className="list-group-item">Division</li>
      <li className="list-group-item">Percent Calculation</li>
    </ul>
    <div className="text-center my-4">
      <Button className="text-white" variant="success" href="/calculator">Try It Now</Button>
    </div>
  </div>
);

export default Home;