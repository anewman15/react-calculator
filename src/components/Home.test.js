/*
 eslint-disable no-unused-vars, import/named, import/no-cycle
*/

import React from 'react';
import { act, render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import pretty from 'pretty';
import Home from './Home';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Home', () => {
  it('should render a Home page with information', () => {
    act(() => {
      render(<Home />, container);

      expect(pretty(container.innerHTML))
        .toMatchSnapshot();
    });
  });
});
