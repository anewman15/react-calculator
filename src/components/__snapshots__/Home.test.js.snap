// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render a Home page with information aboutt the app 1`] = `
<div
  className="container container-fluid"
>
  <h1
    className="text-center mt-4"
  >
    Over the Counter Calculator
  </h1>
  <p
    className="display-5 my-4"
  >
    This is one of the primitive calculators. Built with React. I'll probably improve it over time. Can be used for basic calculations using arithmatic oeprations:
  </p>
  <div
    className="list-group w-25 mx-auto list-group"
    onKeyDown={[Function]}
  >
    <div
      className="list-group-item list-group-item"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Addition
    </div>
    <div
      className="list-group-item list-group-item"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Subtraction
    </div>
    <div
      className="list-group-item list-group-item"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Multiplication
    </div>
    <div
      className="list-group-item list-group-item"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Division
    </div>
    <div
      className="list-group-item list-group-item"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Percent Calculation
    </div>
  </div>
  <div
    className="text-center my-5"
  >
    <a
      className="text-white w-50 btn btn-success"
      href="/calculator"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Try It Now
    </a>
  </div>
</div>
`;
